### PR TITLE
Fix `UnusedException` Error Prone violations

### PR DESCRIPTION
**Pull request by basil**
Fix violations of the [`UnusedException`](https://errorprone.info/bugpattern/UnusedException) Error Prone check, which catches `catch` blocks that catch one exception and re-throw another swallowing the caught exception rather than setting it as a cause. Throwing a new exception without supplying the caught one as a cause means the stack trace will terminate at the `catch` block, which can make debugging harder. This PR propagates the cause in all such cases.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/Failure.java
Comment (daniel-beck) [inline core/src/main/java/hudson/model/Failure.java:52]:
```suggestion
    /**
     * @since TODO
     */
    public Failure(String message, Throwable cause) {
```

Comment (daniel-beck) [reply core/src/main/java/hudson/model/Failure.java:52]: Unsure about this change.

`Failure` is currently (slightly) inconsistent: `Descriptor.FormException` calls an overload of `#generateResponse` that takes a `Throwable` that will be set as request attribute. But why, given the class's Javadoc?

Now, if the idea is the Javadoc doesn't really apply anymore (see `AbstractModelObject/error.jelly` as evidence), would it not make more sense for us to change `#generateResponse` (the other one not taking a `Throwable`) to set the request property if there is a cause from this constructor? And if so, wouldn't we just deprecate `#generateResponse` taking a `Throwable`, and instead change callers to use the constructor taking a `Throwable`?

That would be a change in behavior however, as currently the caller being adapted in this PR doesn't render the exception, but would start doing so if `Failure` sets the `Throwable cause` from the constructor as request property. Perhaps we should have a flag to distinguish causes set for debugging diagnostics only, and causes intended to be rendered?

Probably easier to just skip this one for now.

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
@daniel-beck Does this address your concerns?

**Review by daniel-beck — APPROVED** [review]
Thanks!

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by res0nance — APPROVED** [review]

**Pull request merged**
